src/librustc/hir/mod.rs

```diff
@@ -1905,19 +1905,27 @@ pub struct PolyTraitRef {
 
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
 pub enum Visibility {
-    Public,
-    Crate(CrateSugar),
-    Restricted { path: P<Path>, id: NodeId },
+    Public(Span),
+    Crate(Span, CrateSugar),
+    Restricted { span: Span, path: P<Path>, id: NodeId },
     Inherited,
 }
 
 impl Visibility {
+    pub fn is_pub(&self) -> bool {
+        use self::Visibility::*;
```

Comment:
Nit: Please avoid enum imports and use `Enum::Variant`, this really improves code searchability in rustc.